### PR TITLE
Improve test and remove workarounds

### PR DESCRIPTION
**Pull request by Milly**
## Summary by CodeRabbit

- **New Features**
  - Introduced a new `resetConfig` function to reset application configurations.
  - Enhanced `DenopsImpl` class by changing `call` method to non-async for improved performance.
  - Removed a workaround in `withDenops` function, optimizing for better handling of async operations.

- **Refactor**
  - Refactored `testInternal` function in `tester.ts` to more efficiently handle test attributes and modes.

- **Tests**
  - Added comprehensive test cases for `getConfig()` function behavior under different environment settings.
  - Introduced new test scenarios in `tester_test.ts` and `with_test.ts` to ensure robust error handling and functionality testing.

- **Bug Fixes**
  - Adjusted error handling in `DenopsImpl` to directly throw standard errors, simplifying error management.

### CODE REVIEW

File: denops.ts [hunk 1]
```diff
@@ -24,17 +24,8 @@ export class DenopsImpl implements Denops {
...
-    } catch (err) {
-      // Denops v5 or earlier may throws an error as string in Neovim
-      // so convert the error into an Error instance
-      if (typeof err === "string") {
-        throw new Error(err);
-      }
-      throw err;
-    }
+  call(fn: string, ...args: unknown[]): Promise<unknown> {
+    return this.#client.call("invoke", "call", [fn, ...args]);
```
[the remark this answers is not in the document]
Comment (Milly) [reply denops.ts:28]: `this.#client.call` returns the Promise, so return type is valid.

File: conf.ts [hunk 2]
```diff
@@ -79,3 +79,8 @@ export function getConfig(): Config {
   };
   return conf;
 }
+
+/** @internal for test */
+export function resetConfig(newConf?: Config): void {
+  conf = newConf;
+}
```
Comment (lambdalisue) [inline conf.ts:86]: Please export it like

```ts
export const _internal = { resetConfig };
```

so that user who don't read document can recognize.

**Review by lambdalisue — CHANGES_REQUESTED** [review]
1 nit change request

File: conf.ts (hunk 2 above)
Comment (Milly) [reply conf.ts:86]: `conf.ts` module is not exported in *deno.jsonc*.
So user cannot import it like `import ... from "jsr:@denops/test/conf";`
Will v2.0.0 be released in `deno.land/x`?

Comment (Milly) [reply conf.ts:86]: Even so, it makes sense to add `_internal` since the developer (future me) might make a mistake.

Comment (Milly) [reply conf.ts]: Fixed!

**Review by lambdalisue — APPROVED** [review]
LGTM

**Pull request merged**